### ISSUE TITLE
Feature Request: Support css-free styled components

### ISSUE THREAD

**Issue by darthtrevino** [styled-components/styled-components#1406]
I'd like to use styled-components as a super-terse, lightweight wrapper around an existing CSS framework. I'm not interested in replicating the CSS in my JS code or emitting random class-names with my DOM.  

Imagine you're writing a React layer for something like Bootstrap, and you're inspired by the clean separation from classes and styles that styled-components gives you. You could walk through the component documentation and do things like this:

```jsx
import classnames from 'classnames'

const Alert = styled.div.attrs({
  className: (props) => classnames('alert', { 
    'alert-success': props.success, 
    'alert-info': props.info, 
    'alert-warning': props.warning, 
    'alert-danger': props.danger 
  }),
})

// Later on... 

<Alert success>Hooray!</Alert>
<Alert info>This is an info alert</Alert>

```

The Alert component at the point of defining `attrs` is essentially "styled" since a className attribute is attached. I could add an empty template-tag at the end of this, but it would attach a random class-name to my component, which may be undesirable if you want to author a component library for public use.

tl;dr: The tagged template literal argument should be optional

**Comment by mxstbr** [comment]
While that's an interesting idea, it's very magic and unexpected, and the difference between ` styled.div`` ` and ` styled.div ` isn't big enough to warrant that magic in my opinion. You could probably write a function that does what you're trying to do manually?

**Comment by darthtrevino** [comment]
Sure, and that's basically what I'm doing now.  Imagine a situation where you're making an application by consuming this third party library composed of css-free styled components. Say you want to extend some components, and sprinkle in your own CSS to better match your branding or do something creative. _styled-components_ is has a simple and well known API that can act as a building block to organically grow custom components using library components as a base, and it would be really convenient if the library components adhered to that API instead of some homespun, internal thing.

For the library author, it may be sufficient if _styled-components_ just didn't emit a new stylesheet class when you invoked `styled` with an empty CSS string.

**Comment by mxstbr** [comment]
Oh I see, so what you want is to avoid the empty classes we generate for every component?

**Comment by darthtrevino** [comment]
Exactly, yeah. Those should pop in lazily

**Comment by mxstbr** [comment]
I see what you mean, but unfortunately there's a reason those exist: We need them for component interpolations: https://codesandbox.io/s/k593x7w1w3

```JS
const Button = styled.button`
  color: red;
  // This could be anything that adds more CSS
  ${p => p.primary && 'background: transparent;'}
`

const Wrapper = styled.div`
  ${Button} {
    color: green;
  }
`

const App = () => (
  <Wrapper>
    <Button>Normal</Button>
    <Button primary>Primary</Button>
  </Wrapper>
);
```

Note how the two `Button`s have different class names that have the styling associated with them (`.kuNtvI` and `.hamUYA`), but the Wrapper targets the shared static `.sc-x` classname that's shared between them. (and empty) That means removing the static, component-specific class would break interpolating other styled components. 

Does that make sense?

**Comment by darthtrevino** [comment]
Ahh, I see. I didn't realize that you could do that kind of interpolation.
Thanks for the explanation!

**Comment by switz** [comment]
This totally makes sense; but I don't really use it personally. Would there be a way to disable it and prevent unused classNames? I suppose not, but can't hurt to ask.
